src/geode/geometry/normal_frame_transform.cpp

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2019 - 2025 Geode-solutions
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+#include <geode/basic/logger.hpp>
+#include <geode/basic/pimpl_impl.hpp>
+
+#include <geode/geometry/frame.hpp>
+#include <geode/geometry/normal_frame_transform.hpp>
+#include <geode/geometry/vector.hpp>
+
+namespace geode
+{
+    template < index_t dimension >
+    class NormalFrameTransform< dimension >::Impl
+    {
+    public:
+        Impl( const NormalFrameTransform< dimension >& normal_frame_transform,
+            const Frame< dimension >& from,
+            const Frame< dimension >& to )
+        {
+            for( const auto d_from : LRange{ dimension } )
+            {
+                directions_[d_from] =
+                    normal_frame_transform.direction( d_from );
+                orientations_[d_from] =
+                    normal_frame_transform.orientation( d_from );
```

Comment:
As you show it here, this information is already in the base class -> no need to store it again, just use base class'